src/DotNet.Glob/Evaluation/LetterRangeTokenEvaluator.cs

```diff
@@ -4,19 +4,33 @@ namespace DotNet.Globbing.Evaluation
 {
     public class LetterRangeTokenEvaluator : IGlobTokenEvaluator
     {
+        private readonly bool _caseInsensitive;
         private readonly LetterRangeToken _token;
 
-        public LetterRangeTokenEvaluator(LetterRangeToken token)
+        public LetterRangeTokenEvaluator(LetterRangeToken token, bool caseInsensitive)
         {
+            _caseInsensitive = caseInsensitive;
             _token = token;
         }
+
         public bool IsMatch(string allChars, int currentPosition, out int newPosition)
         {
-            var currentChar = allChars[currentPosition];
             newPosition = currentPosition + 1;
 
             //var currentChar = (char)read;
-            if (currentChar >= _token.Start && currentChar <= _token.End)
+            char start, end, currentChar;
+            start = _token.Start;
+            end = _token.End;
+            currentChar = allChars[currentPosition];
+
+            if (_caseInsensitive)
+            {
+                start = char.ToLowerInvariant(start);
```

Comment:
_token.Start and _token.End don't change, so rather than calling ToLowerInvariant on them for every `Match`, performance wise it might be better to call `ToLowerInvariant` on them once in the constructor and store that for comparison.

Reply:
Agreed. I'll look at this.